adala/skills/collection/prompt_improvement.py

```diff
@@ -0,0 +1,182 @@
+import json
+import logging
+from pydantic import BaseModel, field_validator, Field, ConfigDict, model_validator
+from adala.skills import Skill
+from typing import Any, Dict, List, Optional, Union
+from adala.skills import AnalysisSkill
+from adala.utils.parse import parse_template
+from adala.utils.types import ErrorResponseModel
+
+logger = logging.getLogger(__name__)
+
+
+class PromptImprovementSkillResponseModel(BaseModel):
+
+
+    reasoning: str = Field(..., description="The reasoning for the changes made to the prompt")
+    new_prompt_title: str = Field(..., description="The new short title for the prompt")
+    new_prompt_content: str = Field(..., description="The new content for the prompt")
+
+    # model_config = ConfigDict(
+    #     # omit other fields
+    #     extra="ignore",
+    #     # guard against name collisions with other fields
+    #     populate_by_name=False,
+    # )
+
+    # @field_validator("new_prompt_content", mode="after")
```

Comment:
I'm going to restore this code somehow as we need to post-validate the generating. But the private var `_input_variable` is not working with instructor

Reply:
https://humansignal.atlassian.net/browse/DIA-1489